### PR TITLE
Allow user to specify HTTP and gRPC bind addresses (not just ports.) (Rebase of #153)

### PR DESCRIPTION
**Pull request by ThoreKr**
A rebase of #153

### CODE REVIEW

**Comment by ThoreKr** [comment]
@bboreham could you please review this pr?

File: server/server.go [hunk 1]
```diff
@@ -56,7 +58,9 @@ type Config struct {
 
 // RegisterFlags adds the flags required to config this to the given FlagSet
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	f.StringVar(&cfg.HTTPListenAddress, "server.http-listen-host", "", "HTTP server listen address.")
```
[the remark this answers is not in the document]
Comment (ThoreKr) [reply server/server.go]: Oh, missed that one.

File: server/server_test.go [hunk 2]
```diff
@@ -36,6 +36,9 @@ func (f FakeServer) Succeed(ctx context.Context, req *google_protobuf.Empty) (*g
 func TestErrorInstrumentationMiddleware(t *testing.T) {
 	var cfg Config
 	cfg.RegisterFlags(flag.NewFlagSet("", flag.ExitOnError))
+	cfg.HTTPListenAddress = "localhost"
```
[the remark this answers is not in the document]
Comment (ThoreKr) [reply server/server_test.go]: Should this become a separate test?

[the remark this answers is not in the document]
Comment (ThoreKr) [reply server/server_test.go]: Or should it be tested at all?

[the remark this answers is not in the document]
Comment (ThoreKr) [reply server/server_test.go]: @bboreham what's your opinion on this?

**Comment by Sh4d1** [comment]
@bboreham what's missing to get this merged?

File: server/server_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bboreham) [reply server/server_test.go]: Please make sure the server is tested in its default state.

I don't have a strong opinion whether the new option needs a test.

**Comment by bboreham** [comment]
I think I see two points unresolved.

File: server/server_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ThoreKr) [reply server/server_test.go]: Does this apply to all other changes within this file as well?

[the remark this answers is not in the document]
Comment (bboreham) [reply server/server_test.go]: Just make sure it's tested in its default state.
That's the embarrasing one if it doesn't work.

**Comment by ThoreKr** [comment]
@bboreham and now?

**Comment by bboreham** [comment]
I tell you what, I'll just fix it after.
Thanks for the PR!

File: server/server_test.go [hunk 3]
```diff
@@ -36,6 +36,9 @@ func (f FakeServer) Succeed(ctx context.Context, req *google_protobuf.Empty) (*g
...
+	cfg.HTTPListenPort = 9190
```
[the remark this answers is not in the document]
Comment (bboreham) [reply server/server_test.go]: I suspect it was because port 80 doesn't work for ordinary users.

**Pull request merged**